libs/libc/machine/arch_atomic.c

```diff
@@ -754,3 +754,16 @@ SYNC_VAL_CMP_SWAP(4, uint32_t)
 SYNC_VAL_CMP_SWAP(8, uint64_t)
 
 #endif /* __clang__ */
+
+#ifdef __ghs__
+
+/****************************************************************************
+ * Name: __sync_synchronize
+ ****************************************************************************/
+
+void weak_function __sync_synchronize(void)
```

Comment:
Is this ARM specific? I mean for which platforms you tested it?

Reply:
cortex-m7

Reply:
I mean that maybe implementation should be located somewhere in `libs/libc/machine/arm`?

Reply:
not sure whether it happen on other arch since we just make Greenhill compiler work on cortex-m7 arch so far.

Reply:
Let's keep here for now